Source/Core/VideoBackends/OGL/VertexManager.cpp

```diff
@@ -140,6 +140,15 @@ void VertexManager::vFlush(bool useDstAlpha)
 
 	PrepareDrawBuffers(stride);
 
+	if (!bpmem.genMode.zfreeze && IndexGenerator::GetIndexLen() >= 3)
+	{
+		CalculateZSlope(stride);
```

Comment:
@neobrain , you missunderstand me the path proposed is:
if(!zfreeze)
{
     store_last_triangle_data();
     store_matrix();
}
else
{
        if(!already_calculated)
        {
              calculate_slope();
              update_gpu_slope();
              already_calculated=true;
        }
}
but phire has  point on the memory bandwidth. the best midpoint will be just update_gpu_slope() the fist flush where xfreeze is enabled.